commands/shooter/aimsystem.py

```diff
@@ -0,0 +1,106 @@
+from commands2 import CommandBase
+from wpilib import SmartDashboard
+from wpimath.geometry import Pose2d, Transform2d, Rotation2d
+from operatorinterface import Control2D
+
+from subsystems.shootingsubsystem import ShootingSubsystem
+import constants
+from util.angleoptimize import optimizeAngle
+from util.convenientmath import rotationFromTranslation
+
+
+class AimSystem(CommandBase):
+    def __init__(self, shoot: ShootingSubsystem, shooterOffset: Control2D) -> None:
+        SmartDashboard.putBoolean(constants.kShootingManualModeKey, False)
+        CommandBase.__init__(self)
+        self.setName(__class__.__name__)
+        self.shoot = shoot
+        self.offset = shooterOffset
+        self.addRequirements([self.shoot])
+
+    def execute(self) -> None:
+        if SmartDashboard.getBoolean(constants.kShootingManualModeKey, False):
+            self.manualMode()
+        else:
+            self.normalMode()
+
+    def manualMode(self) -> None:
+        wheelSpeed = SmartDashboard.getNumber(constants.kShootingWheelSpeedKey, 0)
+        hoodAngle = SmartDashboard.getNumber(constants.kShootingHoodAngleKey, 0)
+        turretPosition = SmartDashboard.getNumber(constants.kShootingTurretAngleKey, 0)
+        self.shoot.setWheelSpeed(wheelSpeed)
+        self.shoot.setHoodAngle(Rotation2d.fromDegrees(hoodAngle))
+        self.shoot.rotateTurret(
+            Rotation2d.fromDegrees(turretPosition)
+            + constants.kOffsetAngleRange * self.offset.sideToSide()
+        )
+
+    def normalMode(self) -> None:
+        deltaPos = Pose2d(
+            *[
+                constants.kRobotUpdateRate * i * constants.kPredictiveAimGain
+                for i in SmartDashboard.getNumberArray(
+                    constants.kDriveVelocityKeys, [0, 0, 0]
+                )
+            ]
+        )
+        currentPose = Pose2d(
+            *SmartDashboard.getNumberArray(
+                constants.kRobotPoseArrayKeys.valueKey, [0, 0, 0]
+            )
+        )
```

Comment:
Let's move the dynamic aiming to a separate issue/MR until we are happy with static aiming, since we didn't get much time to test on hardware. Less complex code will be easier to debug.